deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/exception/GenericExceptionMapper.java

```diff
@@ -48,6 +49,21 @@ public Response toResponse(Throwable t) {
                 sb.append("\n").append(sw.toString());
                 return Response.serverError().entity(sb.toString()).build();
             }
+        } else if (t instanceof ConstraintViolationException) {
+            StringBuilder sb = new StringBuilder();
+            if (t.getMessage() != null) {
+                sb.append("\nMessage: ").append(t.getMessage());
+            }
+
+            if (clientError.equals(Constants.CLIENT_ERROR_SHORT)) {
+                return Response.serverError().status(Response.Status.BAD_REQUEST).entity(sb.toString()).build();
+            } else {
+                StringWriter sw = new StringWriter();
+                PrintWriter pw = new PrintWriter(sw);
+                t.printStackTrace(pw);
+                sb.append("\n").append(sw.toString());
+                return Response.serverError().status(Response.Status.BAD_REQUEST).entity(sb.toString()).build();
```

Comment:
check if this says server error with 400 code? In that case you should remove "serverError()"

Reply:
https://user-images.githubusercontent.com/5506038/89687929-f5a2f880-d8b5-11ea-8735-0bacad30f57a.png